src/components/Button/index.js

```diff
@@ -161,94 +161,127 @@ const defaultProps = {
     forwardedRef: undefined,
 };
 
-class Button extends Component {
-    constructor(props) {
-        super(props);
-
-        this.renderContent = this.renderContent.bind(this);
-    }
-
-    componentDidMount() {
-        if (!this.props.pressOnEnter) {
-            return;
+function Button({
+    allowBubble,
+    text,
+    shouldShowRightIcon,
+
+    icon,
+    iconRight,
+    iconFill,
+    iconStyles,
+    iconRightStyles,
+
+    small,
+    large,
+    medium,
+
+    isLoading,
+    isDisabled,
+
+    onPress,
+    onLongPress,
+    onPressIn,
+    onPressOut,
+    onMouseDown,
+
+    pressOnEnter,
+    enterKeyEventListenerPriority,
+
+    style,
+    innerStyles,
+    textStyles,
+
+    shouldUseDefaultHover,
+    success,
+    danger,
+    children,
+
+    shouldRemoveRightBorderRadius,
+    shouldRemoveLeftBorderRadius,
+    shouldEnableHapticFeedback,
+
+    isFocused,
+    nativeID,
+    accessibilityLabel,
+    forwardedRef,
+}) {
+    useEffect(() => {
+        if (!pressOnEnter) {
+            return () => {};
         }
 
         const shortcutConfig = CONST.KEYBOARD_SHORTCUTS.ENTER;
 
         // Setup and attach keypress handler for pressing the button with Enter key
-        this.unsubscribe = KeyboardShortcut.subscribe(
+        return KeyboardShortcut.subscribe(
```

Comment:
Is `KeyboardShortcut.subscribe` call on mount? Shouldn't be more like:
``` 
const unsubscribe = KeyboardShortcut.subscribe(...);

return () => { unsubscribe(); }
```

Reply:
`KeyboardShortcut.subscribe` returns a function to unsubscribe, so I'm just passing it directly to the effect return, [example of other usage](https://github.com/Expensify/App/blob/main/src/components/Pressable/GenericPressable/BaseGenericPressable.js#L128).